### ISSUE TITLE
Cannot reproduce the results of the paper

### ISSUE THREAD

**Issue by runningpp** [Simonhfls/GAPS#1]
we trained the model on CMU datasets followed README.md on Nvidia-RTX-4060Ti  with batchsize=4.
For dress template, we have trained for 3+ days with 1000 epochs. 
For tshirt template, we have trained for 1.5 days with 600 epochs.
During the training, the loss seems good:
![hood-pp](https://github.com/Simonhfls/GAPS/assets/37203710/d7b019af-1dce-4036-a119-d4f9f3919d24)

but both results are terrible, the following pics are test on 07-02 CMU with tshirt template：
![hood-ppp](https://github.com/Simonhfls/GAPS/assets/37203710/22c7ca94-65e1-478b-99a3-b40101e33e12)
![hood-p](https://github.com/Simonhfls/GAPS/assets/37203710/19ef9a62-24ee-4929-a48e-d6569b6b63fe)

Could you tell me how to fix the problem and reproduce the paper results? Thx

**Comment by Simonhfls** [comment]
Hi, thanks for your interest in our work! I retrained the model using a smaller batch size (64), and for reference, here is the output training log after 400 epochs for the T-shirt template:

`
m/Loss: 0.3220 - m/L_Strain: 0.1591 - m/L_Bending: 0.0093 - m/L_Collision: 7.5534e-07 - m/L_Gravity: 0.0413 - m/L_Inertia: 0.0024 - m/L_Isometry: 5.4890e-10 
`

In the training log you provided, it appears that the training has not converged to a good minimum. This might be due to the small batch size used, which can lead to greater instability. I will investigate this issue further when I have time, but here are 2 possible ways I come up with that may help improve the current results:
1. Train on a larger GPU that supports a greater batch size
2. If computational resources are limited, try using a constant learning rate (0.0001). You could modify `lr_scheduler` in the `train_gaps.py` file.

Hope this helps !

**Comment by runningpp** [comment]
> Hi, thanks for your interest in our work! I retrained the model using a smaller batch size (64), and for reference, here is the output training log after 400 epochs for the T-shirt template:
> 
> `m/Loss: 0.3220 - m/L_Strain: 0.1591 - m/L_Bending: 0.0093 - m/L_Collision: 7.5534e-07 - m/L_Gravity: 0.0413 - m/L_Inertia: 0.0024 - m/L_Isometry: 5.4890e-10 `
> 
> In the training log you provided, it appears that the training has not converged to a good minimum. This might be due to the small batch size used, which can lead to greater instability. I will investigate this issue further when I have time, but here are 2 possible ways I come up with that may help improve the current results:
> 
>     1. Train on a larger GPU that supports a greater batch size
> 
>     2. If computational resources are limited, try using a constant learning rate (0.0001). You could modify `lr_scheduler` in the `train_gaps.py` file.
> 
> 
> Hope this helps !

Thanks for this reply. I have solved the problem by your suggestions, but I have met new problem.

I trained the model using 3+ A100 GPUs with batch_size=16, producing good results.
The following pics is a woman with long sleeve (1000iterations):
![gaps_longsleeve](https://github.com/Simonhfls/GAPS/assets/37203710/c639442b-3f83-404b-8d30-4955be1c425a)

But for the pants/shorts template, the results is not good: 
The following pic is a woman with pants (1000 iterations), seems like a falling down pants:
![gaps_pants_1000](https://github.com/Simonhfls/GAPS/assets/37203710/b96628f0-ad0a-4260-845a-000a110b5ebb)

For the pants template, we fond it just ok with less iterations:
The following pic is a woman with pants (400 iterations):
![gaps_pants_400](https://github.com/Simonhfls/GAPS/assets/37203710/52ecc88a-2aa1-4ac5-bd57-d98d741638cd)

The following pic is a man with pants (200 iterations):
![gaps_pants_m200](https://github.com/Simonhfls/GAPS/assets/37203710/a72154b7-de81-455c-b1de-61602846e53d)


But for shorts template, we fond it still not good with less iterations:
The following pic is a man with shorts (200 iterations):
![gaps_shorts_m200](https://github.com/Simonhfls/GAPS/assets/37203710/fe7d2e2d-8521-4cea-b994-29d533d4c8d0)

For the pants/shorts or other kinds, how to fix the falling down pants problems？

**Comment by Simonhfls** [comment]
For bottom garments such as pants or shorts, we address this with an additional 'pinning loss' on the y-coordinates for the set of vertices along the uppermost boundary of the garment. This would prevent the garment from slipping.

**Comment by runningpp** [comment]
> For bottom garments such as pants or shorts, we address this with an additional 'pinning loss' on the y-coordinates for the set of vertices along the uppermost boundary of the garment. This would prevent the garment from slipping.

You are so nice! Thanks very much!

**Comment by runningpp** [comment]
I want to  make the cloth simulations more softly, could you please tell me how to change the STVK parameters or else?

The following video is VTO(Self-Supervised Collision Handling via Generative 3D Garment Models for Virtual Try-On) v.s. GAPS with dress template in CMU_05_05
https://github.com/Simonhfls/GAPS/assets/37203710/9d50dedb-2502-4dbd-9240-880b05c05d07

**Comment by runningpp** [comment]
I want to make the cloth simulations more softly, could you please tell me how to change the STVK parameters or else?

The following video is VTO(Self-Supervised Collision Handling via Generative 3D Garment Models for Virtual Try-On) v.s. GAPS with dress template in CMU_05_05


Uploading dress_0505_cmp.mp4…

**Comment by runningpp** [comment]
I want to make the cloth simulations more softly, could you please tell me how to change the STVK parameters or else?

The following video is [VTO](http://mslab.es/projects/SelfSupervisedGarmentCollisions/) v.s. GAPS with dress template in CMU_05_05

https://github.com/Simonhfls/GAPS/assets/37203710/0bf81d68-8cad-4f2f-9124-26b603163cce

**Comment by runningpp** [comment]
> For bottom garments such as pants or shorts, we address this with an additional 'pinning loss' on the y-coordinates for the set of vertices along the uppermost boundary of the garment. This would prevent the garment from slipping.

Could you please tell me how to set the pining_loss weight?
# Loss weight
w_strain = 1
w_isometry = 2e8
w_bend = 1
w_collision = 250
w_gravity = 1
w_pinning = 1???

**Comment by Simonhfls** [comment]
> > For bottom garments such as pants or shorts, we address this with an additional 'pinning loss' on the y-coordinates for the set of vertices along the uppermost boundary of the garment. This would prevent the garment from slipping.
> 
> You are so nice! Thanks very much!

Hi, no problem :) 
Regarding your questions, adjusting the STVK parameters could indeed help, though we haven't experimented with values different from the current settings yet. Regarding the pinning loss weight, a value of 100 is generally sufficient for pants and shorts. For skirts, a higher value may be necessary.
